syntaxes/sway.tmLanguage.json

```diff
@@ -251,7 +251,7 @@
     },
     {
       "comment": "Type declaration",
-      "begin": "\\b(enum|struct|trait|abi)\\s+([a-zA-Z_][a-zA-Z0-9_]*)",
+      "begin": "\\b(enum|struct|trait|abi|storage)\\s+([a-zA-Z_][a-zA-Z0-9_]*)",
```

Comment:
Instead of adding `storage` here try adding a new segment for type declarations that are nameless like the `top level declarations without names` above it: 
```
  {
      "comment": "Type declaration without name",
      "begin": "\\b(storage)\\s+([a-zA-Z_][a-zA-Z0-9_]*)",
      "end": "[\\{\\(;]",
      "beginCaptures": {
        "1": {
          "name": "storage.type.sway"
        }
      },
      "patterns": [  <-- // TBC I'm unsure of what patterns we want to include for this
        {
          "include": "#block_comment"
        },
        {
          "include": "#line_comment"
        },
        {
          "include": "#type_params"
        },
        {
          "include": "#core_types"
        },
        {
          "include": "#pub"
        },
        {
          "include": "#where"
        }
      ]
    }
```

Reply:
Just tested it out, assuming you want to keep all of the same patterns adding this will give you the desired outcome: 
```
    {
      "comment": "Type declaration without name",
      "begin": "\\b(storage)",
      "end": "[\\{\\(;]",
      "beginCaptures": {
        "1": {
          "name": "storage.type.sway"
        }
      },
      "patterns": [
        {
          "include": "#block_comment"
        },
        {
          "include": "#line_comment"
        },
        {
          "include": "#type_params"
        },
        {
          "include": "#core_types"
        },
        {
          "include": "#pub"
        },
        {
          "include": "#where"
        }
      ]
    }
```

Reply:
@adlerjohn or @sezna I'm assuming we won't need `where` for `storage`?

Reply:
I don't think so, no.